internal/alb/tg/targetgroup_group.go

```diff
@@ -85,17 +88,23 @@ func (controller *defaultGroupController) GC(ctx context.Context, tgGroup Target
 		tagFilters[k] = []string{v}
 	}
 
-	usedTgArns := sets.NewString()
+	usedExternalTGARNs := sets.NewString(tgGroup.externalTGARNs...)
+	usedServiceTGARNs := sets.NewString()
 	for _, tg := range tgGroup.TGByBackend {
-		usedTgArns.Insert(tg.Arn)
+		usedServiceTGARNs.Insert(tg.Arn)
 	}
 	arns, err := controller.cloud.GetResourcesByFilters(tagFilters, aws.ResourceTypeEnumELBTargetGroup)
 	if err != nil {
 		return fmt.Errorf("failed to get targetGroups due to %v", err)
 	}
-	currentTgArns := sets.NewString(arns...)
-	unusedTgArns := currentTgArns.Difference(usedTgArns)
-	for arn := range unusedTgArns {
+	currentServiceTGARNs := sets.NewString(arns...)
+	unusedServiceTGARNs := currentServiceTGARNs.Difference(usedServiceTGARNs)
+	for arn := range unusedServiceTGARNs {
+		if usedExternalTGARNs.Has(arn) {
+			albctx.GetEventf(ctx)(corev1.EventTypeWarning, "Warning", "targetGroup created for k8s service should be referenced by serviceName and servicePort instead of TargetGroupARN: %s", arn)
```

Comment:
Is this something that is actionable, though? The message above doesn't seem to explain *why* this is a warning.

Maybe this should just be an INFO-level log message and perhaps change the message to something like "Ignoring targetGroup XXX during garbage collection because targetGroupARN is set, indicating it was not created by ALB Ingress Controller."?

Reply:
Hi, jay, yes. This action is actionable. If the targetGroup is created by the controller, then user shouldn't use a ARN to reference it. It's stated in docs as well https://kubernetes-sigs.github.io/aws-alb-ingress-controller/guide/ingress/annotation/#actions "use ARN in forward Action".